src/main/java/co/cask/spanner/common/SpannerConfig.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.spanner.common;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Macro;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.hydrator.common.ReferencePluginConfig;
+
+import java.io.IOException;
+import javax.annotation.Nullable;
+
+/**
+ * Spanner configuration
+ */
+public class SpannerConfig extends ReferencePluginConfig {
+  @Name("project")
+  @Description("Project ID")
```

Comment:
descriptions should be more than a name. We need to go through all these plugins and give them better descriptions. In the dataproc provisioner, we use:

Google Cloud Project ID, which uniquely identifies your project. You can find it on the Dashboard in the Cloud Platform Console.